pkg/daemon/certs/certs.go

```diff
@@ -26,26 +26,63 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"reflect"
 	"syscall"
 	"unsafe"
 
 	"golang.org/x/sys/windows"
 	"k8s.io/klog/v2"
 )
 
-// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// previousCABundleFileName tracks all the certs that the operator has imported into the node's local trust store
+const previousCABundleFileName = "imported-certs.pem"
+
+// Reconcile ensures the node's system certificates match the expected state as given by the trusted CA bundle file.
 // Returns a boolean whether any certs needed to be reconciled
-func Reconcile(caBundle string) (bool, error) {
-	if caBundle == "" {
-		// TODO: ensure that all certs that we've added previously are removed https://issues.redhat.com/browse/WINC-688
-		return false, nil
+func Reconcile(caBundlePath string) (bool, error) {
+	var expectedCerts []*x509.Certificate
+	var err error
+	// Read expected certs from CA trust bundle file. Will be empty when a cluster-wide proxy is not in use or removed.
+	if caBundlePath != "" {
+		if expectedCerts, err = readCertsFromFile(caBundlePath); err != nil {
+			return false, fmt.Errorf("Failed to read certificate file %s: %w", caBundlePath, err)
+		}
+		klog.Infof("read %d expected Certs from %s", len(expectedCerts), caBundlePath)
+	}
+
+	// Populate existing certs using the previously imported CA trust bundle.
+	pwd, err := os.Getwd()
+	if err != nil {
+		return false, err
+	}
+	previousCABundlePath := fmt.Sprintf("%s\\%s", pwd, previousCABundleFileName)
```

Comment:
I could just make this a well known location to WICD instead, since it creates/deletes the file. LMK folks

Reply:
Please make this a well known location. In addition please use [filepath-securejoin](https://github.com/cyphar/filepath-securejoin?tab=readme-ov-file#filepath-securejoin). This was suggested to me in one of my upstream PRs.

Reply:
Wouldn't passing in the cert bundle directory via command arguments be the correct thing to do here?

Reply:
I don't see the need as this is a file wholly managed by WICD. WMCO need not know about it